packages/angular_devkit/build_angular/src/dev-server/index.ts

```diff
@@ -214,13 +231,13 @@ export class DevServerBuilder implements Builder<DevServerBuilderOptions> {
         htmlAcceptHeaders: ['text/html', 'application/xhtml+xml'],
       } as WebpackDevServer.HistoryApiFallbackConfig,
       stats: false,
-      compress: browserOptions.optimization,
+      compress: styles || scripts,
       watchOptions: {
         poll: browserOptions.poll,
       },
       https: options.ssl,
       overlay: {
-        errors: !browserOptions.optimization,
+        errors: !(styles || scripts),
```

Comment:
Not sure, why is based on optimisation flag

Reply:
I guess to not show errors on prod builds? But we already heavily discourage the server from being used in prod, since it's not a dev server. So I think it's ok to always be true.